patient-appointment-management/assets/index.html

```diff
@@ -0,0 +1,147 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Patient Appointment Management Application</title>
+  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Roboto:300,300italic,700,700italic">
+  <link rel="stylesheet" href="https://unpkg.com/normalize.css/normalize.css">
+  <link rel="stylesheet" href="https://unpkg.com/milligram/dist/milligram.min.css">
+  <link rel="stylesheet" href="/styles.css">
+</head>
+<body>
+  <header>
+    <h1>Patient Appointment Management Application</h1>
+    <div id="hipaa-disclaimer" style="margin-bottom: 20px;">
+        Note that this application is designed for prototyping purposes only and you should<br/>
+        <a href="https://ahoy.twilio.com/patient-appointment-management" target="_blank">consult with a Twilio Expert</a>
+        before publicly offering this service in a production context.
+    </div>
+    <p>
+        This application is designed for healthcare providers to implement patient appointment reminders.
+        Patients will receive various appointment event such as booking & reminders via SMS
+        and can respond with confirmation and/or cancellations.
+    </p>
+  </header>
+  <main>
+    <section>
+      <div class="flow-loader"></div>
+      <div id="flow-deploy" style="display: none;" class="deployer">
+          <h4>Your Studio Flow is not yet deployed</h4>
+          <a class="button" onclick="deployStudioFlow(event);">Deploy Studio Flow</a>
+          <div id="flow-loader button-loader" style="display: none;"></div>
+      </div>
+      <div id="flow-deployed" style="display: none;" class="deployer">
+          <h4><span class="checkmark">&#10003;</span> Your Studio Flow is deployed</h4>
+          <a id="flow-open" class="button button-outline" href="" target="_blank">Open Studio Flow</a>
+          <p><b>FLOW REST API URL:</b>&nbsp<i id="flow-rest-api-url"></i></p>
+      </div>
+    </section>
+
+    <section>
+      <div class="aws-bucket-loader"></div>
+      <div id="aws-bucket-deploy" style="display: none;" class="deployer">
+        <h4>Your AWS bucket is not yet deployed</h4>
+        <a class="button" onclick="deployAWSBucket(event);">Deploy AWS Bucket</a>
+        <div id="aws-bucket-loader button-loader" style="display: none;"></div>
+      </div>
+      <div id="aws-bucket-deploying" style="display: none;" class="deployer">
+        <h4>Your AWS bucket in being deployed ...</h4>
+        <p>Please periodically refresh this page until deployment is complete</p>
```

Comment:
Is there any way to automate this via `setTimeout()`?

Reply:
I tried sleeping in the client-side java script, but didn't work. For now, leaving it for user will need to refresh the page. Cloudformation deploy takes minutes. If you know of a good javascript to loop/sleep on the client-side can you point me to it? thx